src/hooks/__internal__/useChildButtons/useChildButtons.tsx

```diff
@@ -38,32 +41,28 @@ const useChildButtons = (
 
   useEffect(() => {
     const firstChild = getButtonChildren()?.[0];
-    if (
-      focusFirstChildButtonOnOpen.current &&
-      showAdditionalButtons &&
-      firstChild
-    ) {
-      focusFirstChildButtonOnOpen.current = false;
+    if (focusFirstChildButtonOnOpen && showAdditionalButtons && firstChild) {
+      setFocusFirstChildButtonOnOpen(false);
       firstChild.focus();
     }
-  }, [showAdditionalButtons, getButtonChildren]);
+  }, [showAdditionalButtons, getButtonChildren, focusFirstChildButtonOnOpen]);
 
   const handleToggleButtonKeyDown = (
     ev: React.KeyboardEvent<HTMLButtonElement>
   ) => {
-    if (
+    const isToggleKey =
```

Comment:
Suggestion (non-blocking): I wonder if we could get rid of the useEffect altogether here if we tweak this slightly as currently we're relying on several more updates/effects to trigger this by changing state etc. I think we could remove it and just call the `focus` in the `handleToggleButtonKeyDown` function, we'd likely need a timeout so that we can allow the buttons to be rendered first and so that the query finds them etc

```
  const handleToggleButtonKeyDown = (
    ev: React.KeyboardEvent<HTMLButtonElement>
  ) => {
    const isToggleKey = 
    Events.isEnterKey(ev) ||
    Events.isSpaceKey(ev) ||
    Events.isDownKey(ev) ||
    Events.isUpKey(ev) ||
    (Events.isTabKey(ev) && showAdditionalButtons);
    

    if (isToggleKey) {
      ev.preventDefault();

      if (!showAdditionalButtons) {
        showButtons();
      }

      setTimeout(() => {
        getButtonChildren()?.[0]?.focus();
      })
    }
  };